modules/function/src/main/java/com/opengamma/strata/function/calculation/deposit/TermDepositMeasureCalculations.java

```diff
@@ -114,30 +122,19 @@ private static MultiCurrencyAmount calculatePv01(
       ExpandedTermDeposit product,
       CalculationMarketData marketData) {
 
-    return ratesProviderStream(marketData)
-        .map(provider -> calculateBucketedPv01(product, provider))
-        .collect(toScenarioResult());
+    return FxConvertibleList.of(
```

Comment:
I've remembered the problem with automatically creating `CurrencyValuesArray` and `MultiCurrencyValuesArray` - the return type of the function would have to be `ScenarioResult<CurrencyAmount>` instead of `CurrencyValuesArray`. Whether or not that's a problem is debatable.